src/components/navBar/navBar.js

```diff
@@ -0,0 +1,496 @@
+    }
+  }
+
+  /**
+   * Move focus from oldTab to newTab.
+   * @param {!NavLinkController} oldTab
+   * @param {!NavLinkController} newTab
+   */
+  function moveFocus(oldTab, newTab) {
+    oldTab.setFocused(false);
+    newTab.setFocused(true);
+  }
+
+  /**
+   * Responds to keypress events.
+   * @param {!Event} e
+   */
+  function onKeydown(e) {
+    var keyCodes = $mdConstant.KEY_CODE;
+    var tabs = getTabs();
+    var focusedTab = getFocusedTab();
+    if (!focusedTab) return;
+
+    var focusedTabIndex = tabs.indexOf(focusedTab);
+
+    // use arrow keys to navigate between tabs
+    switch (e.keyCode) {
+      case keyCodes.UP_ARROW:
+      case keyCodes.LEFT_ARROW:
+        if (focusedTabIndex > 0) {
+          moveFocus(focusedTab, tabs[focusedTabIndex - 1]);
+        }
+        break;
+      case keyCodes.DOWN_ARROW:
+      case keyCodes.RIGHT_ARROW:
+        if (focusedTabIndex < tabs.length - 1) {
+          moveFocus(focusedTab, tabs[focusedTabIndex + 1]);
+        }
+        break;
+      case keyCodes.SPACE:
+      case keyCodes.ENTER:
+        // timeout to avoid a "digest already in progress" console error
+        $timeout(function() {
+          focusedTab.getButtonEl().click();
+        });
+        break;
+    }
+  }
+}
+
+function MdNavLink() {
+  return {
+    restrict: 'E',
+    require: '^MdNavBar',
+    controller: MdNavLinkController,
+    bindToController: true,
+    controllerAs: 'ctrl',
+    replace: true,
+    transclude: true,
+    template:
+      '<li class="md-nav-link" role="option" aria-selected="{{ctrl.isSelected()}}">' +
+        '<md-button ng-if="ctrl.navSref" class="md-nav-button md-accent"' +
+          'ng-class="ctrl.getNgClassMap()"' +
+          'tabindex="-1"' +
+          'ui-sref="{{ctrl.navSref}}">' +
+          '<span ng-transclude></span>' +
+        '</md-button>' +
+        '<md-button ng-if="ctrl.navHref" class="md-nav-button md-accent"' +
+          'ng-class="ctrl.getNgClassMap()"' +
+          'tabindex="-1"' +
+          'ng-href="{{ctrl.navHref}}">' +
+          '<span ng-transclude></span>' +
+        '</md-button>' +
+        '<md-button ng-if="ctrl.navClick" class="md-nav-button md-accent"' +
+          'ng-class="ctrl.getNgClassMap()"' +
+          'tabindex="-1"' +
+          'ng-click="ctrl.navClick()">' +
+          '<span ng-transclude></span>' +
+        '</md-button>' +
+      '</li>',
+    scope: {
+      'navClick': '&?',
+      'navHref': '@?',
+      'navSref': '@?',
+      'linkName': '@',
+    }
+  };
+}
+
+/**
+ * Controller for the nav-link component.
+ * @param {!angular.JQLite} $element
+ * @constructor
+ * @final
+ * @ngInject
+ */
+function MdNavLinkController($element) {
+  var ctrl = this;
+
+  // Data-bound variables
+  /** @export @const {?Function} */
+  ctrl.navClick;
+  /** @export @const {?string} */
+  ctrl.navHref;
+  /** @export @const {?string} */
+  ctrl.navSref;
+
+
+  // State variables
+  /** @private {boolean} */
+  var selected_ = false;
```

Comment:
Inconsistent use of leading vs. training underscore in this file.

Also, these aren't _properties_, so technically they can't be `@private`